internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -46,7 +46,7 @@ import (
 // sync is responsible for reconciling deployments on scaling events or when they
 // are paused.
 func (r *Reconciler) sync(ctx context.Context, md *clusterv1.MachineDeployment, msList []*clusterv1.MachineSet) error {
-	newMS, oldMSs, err := r.getAllMachineSetsAndSyncRevision(ctx, md, msList, false)
+	newMS, oldMSs, err := r.getAllMachineSetsAndSyncRevision(ctx, md, msList, false, false)
```

Comment:
Are we intentionally setting this always to false? 

I guess the answer might be yes, but I'm not sure how we can ensure the same reasoning will apply going forward. Should we maybe just pass in the correct information? (as far as I can see we have it available where we call sync)

Reply:
We have that information 3 lines below where we call sync. Now we are hard-coding that the templates don't exist whenever this func is called from sync. I think it can easily happen that code changes and we run into problems because of this wrong information here

Reply:
fixed